File: src/main/java/com/github/xumeng/java/problems/easy/problem0303/NumArray0303.java
package com.github.xumeng.java.problems.easy.problem0303;

public class NumArray0303 {

    private final int[] nums;


    public NumArray0303(int[] nums) {
        this.nums = nums;
    }

    public int sumRange(int left, int right) {
        int sum = 0;
        for (int i = left; i <= right; i++) {
            sum += nums[i];
        }
        return sum;
    }

}

File: src/main/kotlin/com/github/xumeng/kotlin/problems/easy/problem0303/NumArray0303.kt
package com.github.xumeng.kotlin.problems.easy.problem0303

class NumArray0303(nums: IntArray) {

    private var numArray = nums

    fun sumRange(left: Int, right: Int): Int {
        var sum = 0
        for (i in left..right) {
            sum += numArray[i]
        }
        return sum
    }

}